src/main/scala/org/broadinstitute/hail/expr/AST.scala

```diff
@@ -939,11 +939,26 @@ case class ApplyMethod(posn: Position, lhs: AST, method: String, args: Array[AST
           parseError(s"method `$method' expects a lambda function [param => Boolean], got [param => ${body.`type`}]")
         `type` = TBoolean
 
+      case (arr: TArray, "sort", rhs) =>
+        lhs.typecheck(ec)
+        rhs match {
+          case Array() =>
+          case Array(Const(_, asc, _)) =>
+            if (!asc.isInstanceOf[Boolean])
+              parseError(s"optional parameter for method `$method' must be `true' for ascending (default) or `false' for descending, got `$asc'")
```

Comment:
What about "Second (optional) argument" instead of "optional paramter".  Arguments are what callers pass to functions.  Parameters are names of variables in called functions to which arguments are bound.  `def f(i: Int) = ...; f(5)`.  `i` is a parameter.  `5` is an argument.